ci-operator/step-registry/ipi/conf/ibmcloud/kms-key/ipi-conf-ibmcloud-kms-key-commands.sh

```diff
@@ -6,10 +6,79 @@ set -o pipefail
 
 CONFIG="${SHARED_DIR}/install-config.yaml"
 
-cat "${SHARED_DIR}/ibm_kpKey.yaml"
+echo "ControlPlane EncryptionKey: ${IBMCLOUD_CONTROL_PLANE_ENCRYPTION_KEY}"
+echo "Compute EncryptionKey: ${IBMCLOUD_COMPUTE_ENCRYPTION_KEY}"
+echo "DefaultMachinePlatform EncryptionKey: ${IBMCLOUD_DEFAULT_MACHINE_ENCRYPTION_KEY}"
 
-yq-go m -x -i "${CONFIG}" "${SHARED_DIR}/ibm_kpKey.yaml"
+key_file="${SHARED_DIR}/ibmcloud_key.json"
+if [[ ! -f "${key_file}" ]]; then
+    echo "ERROR: fail to find the key info file ${key_file} !!"
+    exit 1
+fi
+cat ${key_file}
 
-cat "${CONFIG}"
+resource_group=$(jq -r .resource_group ${key_file})
 
+# Set EncryptionKey for control plane nodes
+CONFIG_PATCH="${SHARED_DIR}/install-config-ibmcloud-kpkey.yaml.patch"
+if [[ "${IBMCLOUD_CONTROL_PLANE_ENCRYPTION_KEY}" == "true" ]]; then
+    crn_master=$(jq -r .master.keyCRN ${key_file})
+    if [[ -z ${crn_master} ]]; then
+        echo "ERROR: fail to get the crn info of the master key in ${key_file} !!"
+        exit 1
+    fi
+    cat >> "${CONFIG_PATCH}" << EOF
+controlPlane:
+  platform:
+    ibmcloud:
+      bootVolume:
+        encryptionKey: "${crn_master}"
+EOF
+fi
 
+#Set EncryptionKey for compute nodes
+if [[ "${IBMCLOUD_COMPUTE_ENCRYPTION_KEY}" == "true" ]]; then
+    crn_worker=$(jq -r .worker.keyCRN ${key_file})
+    if [[ -z ${crn_worker} ]]; then
+        echo "ERROR: fail to get the crn info of the worker key in ${key_file} !!"
+        exit 1
+    fi
+    cat >> "${CONFIG_PATCH}" << EOF
+compute:
+- platform:
+    ibmcloud:
+      bootVolume:
+        encryptionKey: "${crn_worker}"
+EOF
+fi
+
+# Set EncryptionKey under defaultMachinePlatform, applied to all nodes
+if [[ "${IBMCLOUD_DEFAULT_MACHINE_ENCRYPTION_KEY}" == "true" ]]; then
+    crn_default=$(jq -r .default.keyCRN ${key_file})
+    if [[ -z ${crn_default} ]]; then
+        echo "ERROR: fail to get the crn info of the default key in ${key_file} !!"
+        exit 1
+    fi
+    cat >> "${CONFIG_PATCH}" << EOF
+platform:
+  ibmcloud:
+    defaultMachinePlatform:
+      bootVolume:
+        encryptionKey: "${crn_default}"
+EOF
+fi
+
+cat >> "${CONFIG_PATCH}" << EOF
+platform:
+  ibmcloud:
+    resourceGroupName: ${resource_group}
+EOF
+
+if [[ -f "${CONFIG_PATCH}" ]]; then
```

Comment:
When calling the step, users is enabling kms key testing on purpose, I do not think we need to check the file exists or not.